truss/templates/server/model_wrapper.py

```diff
@@ -39,28 +100,39 @@ def load(self):
         if _signature_accepts_keyword_arg(model_class_signature, "secrets"):
             model_init_params["secrets"] = SecretsResolver.get_secrets(self._config)
         self._model = model_class(**model_init_params)
+
         if hasattr(self._model, "load"):
             self._model.load()
-        self.ready = True
 
-    def preprocess(self, request: Dict) -> Dict:
+    async def preprocess(
+        self,
+        payload: Union[Dict, CloudEvent, ModelInferRequest],
+        headers: Dict[str, str] = None,
+    ) -> Union[Dict, ModelInferRequest]:
         if not hasattr(self._model, "preprocess"):
-            return request
-        return self._model.preprocess(request)
+            return payload
+        return self._model.preprocess(payload)
```

Comment:
Should we acquire `self._predict_lock` here before calling pre-processing which might also touch GPU memory? The issue here is that not acquiring the local for the whole request, might results in one request preprocessing after another was preprocessed but not predicted, which might be an issue in the GPU memory case.

Reply:
Is this something we guarantee? I wonder if we should explicitly not support this in the pre-process method.  I would like to keep things async in pre-process in case they involve I/O.  Any examples we can go off of for models that might do this?